test/instrumentation/modules/http/disabling.js

```diff
@@ -0,0 +1,162 @@
+      t.ok(/GET localhost:\d+\//.test(span.name), 'span name')
+      t.equal(span.type, 'ext.http.http', 'span type')
+    }
+
+    t.test('incoming enabled + outgoing enabled', makeTest({
+      disableInstrumentations: '',
+      instrumentIncomingHTTPRequests: true
+    }, (t, data) => {
+      t.equal(data.transactions.length, 2, 'transaction count')
+      t.equal(data.spans.length, 1, 'span count')
+
+      assertRequestTransaction(t, data.transactions)
+      assertTopTransaction(t, data.transactions)
+      assertSpan(t, data.spans[0])
+
+      t.end()
+    }))
+
+    t.test('incoming enabled + outgoing disabled', makeTest({
+      disableInstrumentations: 'http',
+      instrumentIncomingHTTPRequests: true
+    }, (t, data) => {
+      t.equal(data.transactions.length, 2, 'transaction count')
+      t.equal(data.spans.length, 0, 'span count')
+
+      assertRequestTransaction(t, data.transactions)
+      assertTopTransaction(t, data.transactions)
+
+      t.end()
+    }))
+
+    t.test('incoming disabled + outgoing enabled', makeTest({
+      disableInstrumentations: '',
+      instrumentIncomingHTTPRequests: false
+    }, (t, data) => {
+      t.equal(data.transactions.length, 1, 'transaction count')
+      t.equal(data.spans.length, 1, 'span count')
+
+      assertTopTransaction(t, data.transactions)
+      assertSpan(t, data.spans[0])
+
+      t.end()
+    }))
+
+    t.test('incoming disabled + outgoing disabled', makeTest({
+      disableInstrumentations: 'http',
+      instrumentIncomingHTTPRequests: false
+    }, (t, data) => {
+      t.equal(data.transactions.length, 1, 'transaction count')
+      t.equal(data.spans.length, 0, 'span count')
+
+      assertTopTransaction(t, data.transactions)
+
+      t.end()
+    }))
+  })
+} else {
+  const http = require('http')
+
+  process.on('message', config => {
+    class MockTransport {
+      constructor () {
+        this.transactions = []
+        this.spans = []
+      }
+
+      config () { }
+
+      sendSpan (span) {
+        this.spans.push(span)
+      }
+
+      sendTransaction (transaction) {
+        this.transactions.push(transaction)
+      }
+    }
+
+    const mock = new MockTransport()
+
+    const agent = require('../../../..').start(Object.assign({
+      captureExceptions: false,
+      metricsInterval: 0,
+      centralConfig: false,
+      transport: () => mock
+    }, config))
+
+    const express = require('express')
+
+    const app = express()
+
+    app.get('/', (req, res) => {
+      res.end('hello')
+    })
+
+    let trans
+    sendRequest(app).then(() => {
+      trans.end()
+      setTimeout(() => {
+        process.send(mock)
+      }, 10)
+    })
+
+    function ping (url, fn = res => res.resume()) {
+      return new Promise((resolve, reject) => {
+        const req = http.get(url, res => {
+          res.on('error', reject)
+          res.on('end', resolve)
+          fn(res)
+        })
+        req.on('error', reject)
+        req.end()
```

Comment:
No need to call `end()` on a `http.get()` request as they are automatically ended